Emby.Server.Implementations/Library/UserDataManager.cs

```diff
@@ -237,7 +243,10 @@ private static string GetCacheKey(long internalUserId, Guid itemId)
         /// <inheritdoc />
         public UserItemData? GetUserData(User user, BaseItem item)
         {
-            return GetUserData(user, item.Id, item.GetUserDataKeys());
+            return item.UserData.Where(e => e.UserId.Equals(user.Id)).Select(Map).FirstOrDefault() ?? new UserItemData()
```

Comment:
The original query pulls the UserData that matches the ItemId first and then falls back to a random key. It probably doesn't matter but thought I would point it out

Reply:
yea i dont think that matters. This is already matched by id first by being loaded as a foreignkey. In my testing that did not make any difference but i agree we should keep that in mind if issues arise.

Reply:
yea so i double checked, the CustomDataKey is still a relict that was important in the pre-foregin key times where the connection between UserData and Item was not relational. Now its only important if we want to reconnect detached UserData. For the actual act of reading the data it makes little difference as long as we always provide _the same_ which we do via FirstOrDefault which coincidentially should always be the one thats the ID one anyway which mimics the behavior as before.